enterprise/api/v1/views.py

```
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
+
+        Note: This endpoint isn't transactional in that if an enrollment fails, there can potentially be a scenario
+        with a subset of users enrolled. Requested learners' eligibility to enroll are expected to be pre-vetted.
+
+        Expected params:
+            - license_info (Dict): user emails as keys and the license UUIDs associated with each of the courses the
+            user's being enrolled in as values. Note, each user is expected to have exactly one license per course
+            listed in the 'courses' parameter.
+            Example:
+                license_info: {
+                    'newuser@test.com': {
+                        'course-v1:edX+DemoX+Demo_Course': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                }
+
+            - courses (Dict): course run keys as keys and the course mode as the value.
+            Example:
+                'courses': {
+                    'course-v1:edX+DemoX+Demo_Course': 'verified',
+                    ...
+                }
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        if serializer.is_valid(raise_exception=True):
+            errors = []
+            enrolled_count = 0
+            license_uuids = serializer.validated_data.get('license_info')
+            course_run_keys = serializer.validated_data.get('courses')
+            users_missing_licenses = []
+
+            # Get a list of user emails from the license info.
+            emails = set(license_uuids.keys())
+            try:
+                for email in emails:
+                    try:
+                        validate_email_to_link(email, enterprise_customer, raise_exception=False)
```

If this is raise_exception is false, where do you get an error? Will this be silently failing on us? Does it return the errors in a list we aren't reading back?

---

Alternately, since you wrap it in a try/except, change the flag.

---

FWIW the email validation logic here is copied from the original bulk enrollment endpoint.